numpy/core/src/multiarray/ctors.c

```diff
@@ -1976,18 +1976,32 @@ PyArray_FromInterface(PyObject *origin)
                     "(data pointer integer, read-only flag)");
             goto fail;
         }
+
         dataptr = PyTuple_GET_ITEM(attr, 0);
-        if (PyString_Check(dataptr)) {
-            res = sscanf(PyString_AsString(dataptr),
-                         "%p", (void **)&data);
+        if (PyBytes_Check(dataptr)) {
+            res = sscanf(PyBytes_AsString(dataptr), "%p", (void **)&data);
             if (res < 1) {
                 PyErr_SetString(PyExc_TypeError,
                         "__array_interface__ data string cannot be converted");
                 goto fail;
             }
         }
+        else if (PyUnicode_Check(dataptr)) {
+            PyObject *tmp = PyLong_FromUnicodeObject(dataptr, 0);
```

Comment:
This is totally new behavior, and it doesn't seem useful. I think I discussed this elsewhere.

Reply:
Also save me needing to write a release note.